src/Symfony/Component/String/AbstractString.php

```diff
@@ -0,0 +1,737 @@
+    {
+        $str = clone $this;
+        $str->string = mb_strtolower(str_replace('İ', 'i̇', $str->string), 'UTF-8');
+
+        return $str;
+    }
+
+    public function match(string $pattern, int $flags = 0, int $offset = 0): ?array
+    {
+        if ('g' === $pattern[-1]) {
+            $pattern = substr($pattern, 0, -1);
+            $match = 'preg_match_all';
+        } else {
+            $match = 'preg_match';
+        }
+
+        if ($this->ignoreCase) {
+            $pattern .= 'i';
+        }
+
+        set_error_handler(static function ($t, $m) { throw new InvalidArgumentException($m); });
+
+        try {
+            if (false === $ret = $match($pattern.'u', $this->string, $matches, $flags, $offset)) {
+                $lastError = preg_last_error();
+
+                foreach (get_defined_constants(true)['pcre'] as $k => $v) {
+                    if ($lastError === $v && '_ERROR' === substr($k, -6)) {
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        return $ret ? $matches : null;
+    }
+
+    /**
+     * @return static
+     */
+    public function padBoth(int $length, string $padStr = ' '): self
+    {
+        if ('' === $padStr || !preg_match('//u', $padStr)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $pad = clone $this;
+        $pad->string = $padStr;
+
+        return $this->pad($length, $pad, STR_PAD_BOTH);
+    }
+
+    /**
+     * @return static
+     */
+    public function padEnd(int $length, string $padStr = ' '): self
+    {
+        if ('' === $padStr || !preg_match('//u', $padStr)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $pad = clone $this;
+        $pad->string = $padStr;
+
+        return $this->pad($length, $pad, STR_PAD_RIGHT);
+    }
+
+    /**
+     * @return static
+     */
+    public function padStart(int $length, string $padStr = ' '): self
+    {
+        if ('' === $padStr || !preg_match('//u', $padStr)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $pad = clone $this;
+        $pad->string = $padStr;
+
+        return $this->pad($length, $pad, STR_PAD_LEFT);
+    }
+
+    /**
+     * @return static
+     */
+    abstract public function prepend(string ...$prefix): self;
+
+    /**
+     * @return static
+     */
+    public function repeat(int $multiplier): self
+    {
+        $str = clone $this;
+        $str->string = str_repeat($str->string, $multiplier < 0 ? 0 : $multiplier);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    abstract public function replace(string $from, string $to): self;
+
+    /**
+     * @param string|callable $to
```

Comment:
The `is_array` in the function indicates to me that `$to` can also be an array.

Also, on other implementations of this method there is no doc comment.

Reply:
that's covered by `callable`